wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +456,143 @@ def get_device_unique_id(device):
 
     def disconnect(self):
         self.logger.info("LenovoSystem disconnected")
+
+    def __getattr__(self, attr):
+        """
+        Tries to delegate a not found method to objects inside self._delegate
+        :raises AttributeError if ``attr`` is not found in any objects this should delegate
+        """
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                return getattr(delegate_to, attr)
+        raise AttributeError(attr)
+
+
+class LenovoSwitches(object):
+    """ Encapsulates all Lenovo Switches behavior """
+    API_PATH = "switches"
+
+    _switches_inventory_available = {
+        'product_name': lambda self, requester: self.get_switch_product_name(requester.name),
+        'uuid': lambda self, requester: self.get_switch_uuid(requester.name),
+        'ports': lambda self, requester: self.get_switch_ports(requester.name),
+        'power_state': lambda self, requester: self.get_switch_power_status(requester.name),
+        'health_state:': lambda self, requester: self.get_switch_health_state(requester.name),
+        'type': lambda self, requester: self.get_switch_type(requester.name),
+        'ipv4_addresses': lambda self, requester: self.get_switch_ipv4_addresses(requester.name),
+        'ipv6_addresses': lambda self, requester: self.get_switch_ipv6_addresses(requester.name),
+        'manufacturer': lambda self, requester: self.get_switch_manufacturer(requester.name),
+        'description': lambda self, requester: self.get_switch_description(requester.name),
+        'serial_number': lambda self, requester: self.get_switch_serial_number(requester.name),
+        'part_number': lambda self, requester:
+            self.get_switch_part_number(requester.name),
+        'ipv4_assignments': lambda self, requester:
+            self.get_switch_ipv4_assignments(requester.name),
+        'ipv6_assignments': lambda self, requester:
+            self.get_switch_ipv6_assignments(requester.name),
+        'firmwares': lambda self, requester: self.get_switch_firmwares(requester.name),
+    }
+
+    def __init__(self, service_instance):
+        self._service_instance = service_instance
+
+    def switch_inventory(self, requester=None, *requested_items, **kwargs):
+
+        # Retrieve and return the switch inventory
+        requested_items = requested_items or self._switches_inventory_available
+
+        return {item: self._switches_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def _list_switches(self):
+        # Collect the switches associated with this LXCA
+        self._switches_list = self._service_instance["switchList"]
+        return self._service_instance["switchList"]
+
+    @property
+    def switches_list(self):
+        if not self._switches_list:
```

Comment:
This creates a bit of an odd pattern here, and I don't see the need to split `_list_switches` into its own method.
I think `self._switches_list` should be set directly here and then returned.